### ISSUE TITLE
[BUG] 403 on `npm whoami` or `npm adduser`

### ISSUE THREAD

**Issue by jgfidelis** [npm/cli#1716]
This might not be a bug, don't  know how to ask for help on this, but it is really weird.

### Current Behavior:
When I enter npm whoami, npm adduser, npm logout, anything; I get a 403 error.

It started out of nowhere, I don't know what caused. I already updated my macOS to the latest version and NPM to the latest version as well.

```
~ npm whoami --verbose
npm info it worked if it ends with ok
npm verb cli [
npm verb cli   '/usr/local/Cellar/node/14.8.0/bin/node',
npm verb cli   '/usr/local/bin/npm',
npm verb cli   'whoami',
npm verb cli   '--verbose'
npm verb cli ]
npm info using npm@6.14.7
npm info using node@v14.8.0
npm verb npm-session 5cdabea0e95d570a
npm http fetch GET 403 http://registry.npmjs.org/-/whoami 134ms
npm verb stack Error: 403 Forbidden - GET http://registry.npmjs.org/-/whoami
npm verb stack     at /usr/local/lib/node_modules/npm/node_modules/npm-registry-fetch/check-response.js:114:15
npm verb stack     at processTicksAndRejections (internal/process/task_queues.js:93:5)
npm verb statusCode 403
npm verb cwd /Users/jgfidelis/Development/xxx
npm verb Darwin 19.6.0
npm verb argv "/usr/local/Cellar/node/14.8.0/bin/node" "/usr/local/bin/npm" "whoami" "--verbose"
npm verb node v14.8.0
npm verb npm  v6.14.7
npm ERR! code E403
npm ERR! 403 403 Forbidden - GET http://registry.npmjs.org/-/whoami
npm ERR! 403 In most cases, you or one of your dependencies are requesting
npm ERR! 403 a package version that is forbidden by your security policy.
npm verb exit [ 1, true ]
npm timing npm Completed in 425ms

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jgfidelis/.npm/_logs/2020-08-24T19_48_37_914Z-debug.log
~ npm logout --verbose
npm info it worked if it ends with ok
npm verb cli [
npm verb cli   '/usr/local/Cellar/node/14.8.0/bin/node',
npm verb cli   '/usr/local/bin/npm',
npm verb cli   'logout',
npm verb cli   '--verbose'
npm verb cli ]
npm info using npm@6.14.7
npm info using node@v14.8.0
npm verb npm-session 06909275e0abbe46
npm verb logout clearing session token for http://registry.npmjs.org/
npm http fetch DELETE 403 http://registry.npmjs.org/-/user/token/d765232f-c81c-406d-bfba-17d9831ea0ed 122ms
npm verb stack Error: 403 Forbidden - DELETE http://registry.npmjs.org/-/user/token/d765232f-c81c-406d-bfba-17d9831ea0ed
npm verb stack     at /usr/local/lib/node_modules/npm/node_modules/npm-registry-fetch/check-response.js:114:15
npm verb stack     at processTicksAndRejections (internal/process/task_queues.js:93:5)
npm verb statusCode 403
npm verb cwd /Users/jgfidelis/Development/xxx
npm verb Darwin 19.6.0
npm verb argv "/usr/local/Cellar/node/14.8.0/bin/node" "/usr/local/bin/npm" "logout" "--verbose"
npm verb node v14.8.0
npm verb npm  v6.14.7
npm ERR! code E403
npm ERR! 403 403 Forbidden - DELETE http://registry.npmjs.org/-/user/token/d765232f-c81c-406d-bfba-17d9831ea0ed
npm ERR! 403 In most cases, you or one of your dependencies are requesting
npm ERR! 403 a package version that is forbidden by your security policy.
npm verb exit [ 1, true ]
npm timing npm Completed in 693ms

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/jgfidelis/.npm/_logs/2020-08-24T19_48_44_347Z-debug.log
```

### Expected Behavior:
npm whoami and related works

### Environment:
- OS: macOS Catalina 10.15.6
- Node: v14.8.0
- npm: 6.14.7

**Comment by jgfidelis** [comment]
I already tried uninstalling & reinstalling npm.

**Comment by ljharb** [comment]
What does `type whoami` or `alias whoami` print out?

**Comment by jgfidelis** [comment]
hey @ljharb 

```
~ type whoami
whoami is /usr/bin/whoami
~ alias whoami
alias: Body cannot be empty
```

**Comment by jgfidelis** [comment]
Well some other data...
I can confirm that the same credentials work and I can login on the npmjs.com site. Same login, password and email. Yet npm adduser does not work.

I have deleted the `~/.npmrc` file, and now `npm whoami` throws `ENEEDAUTH` error but `npm adduser` still does not work with 403 error

**Comment by jgfidelis** [comment]
Hey guys, I found out.

Looks like my IP was in some kind of blacklist.

We found out by intercepting the `npm adduser` requests with a proxy and seeing the results. The first POST to login that returned 403, contained in the body a html with a captcha from cloudflare.

I then googled it and found this thread, and through a website, found out my IP was blacklisted for some reason.

https://community.cloudflare.com/t/getting-lots-of-captcha-since-01-21-2019/57266

I then reset my moden, got a new IP and everything worked.

Thanks.
